tests/tests/cross-mount-inode-del.sh

```diff
@@ -0,0 +1,34 @@
+#
+# Test basic correctness of inode deletion behavior across nodes
+#
+
+t_require_mounts 2
+
+echo "== open a file and delete it cross-mount"
+# write some file contents on mount 0
+echo "hellothere" > "$T_D0/file"
+# hold a read file descriptor open on mount 0
+exec 42< "$T_D0/file"
+# unlink the file on mount 1
+rm -f "$T_D1/file"
+t_sync_seq_index
+
+# verify inode is not yet deletable
+echo "orphan list present $(cat $(t_sysfs_path 1)/forest/orphan_list_present_hint)"
+
+# make sure we can still read through fd on mount 0
+# This actually doesn't seem to close fd.
+cat <&42
+# close it for real
+exec 42>&-
+
+# now deletable but we don't know it until the orphans are processed again
+echo "orphan list present $(cat $(t_sysfs_path 1)/forest/orphan_list_present_hint)"
```

Comment:
Just an observation: I don't think there's anything stopping the orphan list from being processed before we cat the hint after closing.  It'd be really hard to hit but we could get spurious test failures from it.  Something to try and remember and keep an eye out for, I guess.

Reply:
I hear that.